### ISSUE TITLE
Timeout error message should distinguish the transport used

### ISSUE THREAD

**Issue by glennsarti** [puppetlabs/bolt#1307]
## Describe the Bug
I am writing an invetory plugin which returned a uri of `hostname` (e.g. `dc01.bolt.local`).  It seems that bolt defaults to ssh for the transport where the protocol is not specified.  However in this instance I wanted to connect via WinRM.

However the debug (and --trace) messages did not show that it was using the SSH transport instead of WinRm.
```
{"node":"CN=DC01,OU=Domain Controllers,DC=bolt,DC=local","target":"CN=DC01,OU=Domain Controllers,DC=bolt,DC=local","action":null,"object":null,"status":"failure","result":{"_error":{"kind":"puppetlabs.tasks/connect-error","msg":"Timeout after 10 seconds connecting to CN=DC01,OU=Domain Controllers,DC=bolt,DC=local","details":{},"issue_code":"CONNECT_ERROR"}}}
Failed on CN=DC01,OU=Domain Controllers,DC=bolt,DC=local:
  Timeout after 10 seconds connecting to CN=DC01,OU=Domain Controllers,DC=bolt,DC=local
Finished: command 'Write-host 'Hello'' with 2 failures in 413.73 sec
Failed on 2 nodes: CN=DC01,OU=Domain Controllers,DC=bolt,DC=local,CN=TestComputer,OU=Bolt Computers,DC=bolt,DC=local
Ran on 2 nodes in 6 min, 54 sec
```
Both the WinRM and SSH transport emit exactly the same debug message for the timeout.  This is very confusing as the only way I knew what was going on was to use pry into the Bolt source code which is a very poor user experience

Also the debug message only shows the target name, not the target 'uri' so debugging is also going to be difficult

## Expected Behavior
The error message, at least in debug mode, should emit WHAT transport emitted the error message

## Steps to Reproduce
Use bolt to connect to a Windows machine but use ssh

## Environment
 - Version Bolt 1.33.0 (Gem)
 - Platform Windows 10 1903

**Comment by adreyer** [comment]
The simplest solution to this is probably to add the transport type to each targets logger

**Comment by lucywyman** [comment]
> The simplest solution to this is probably to add the transport type to each targets logger
We should instead add a debug message saying what the transport is when we connect to a node.

**Comment by lucywyman** [comment]
Closed by #1323
